File: 0821-shortest-distance-to-a-character/0821-shortest-distance-to-a-character.cpp
class Solution {
public:
    vector<int> shortestToChar(string s, char c) {
        vector<int> res;
        int size=s.size();
        int position=size+1;
        for(int i=0;i<size;i++)
        {
            if(s[i]==c)
            {
                position=i;
            }
            res.push_back(abs(position-i));
        }
        for(int i=position;i>=0;i--)
        {
            if(s[i]==c)
            {
                position=i;
            }
            res[i]=min(abs(position-i),res[i]);
        }
        return res;
    }
};

File: 0821-shortest-distance-to-a-character/0821-shortest-distance-to-a-character.py
class Solution:
    def shortestToChar(self, s: str, c: str) -> List[int]:
        left=-1
        right=-1
        nums=[0]*len(s)

        for i in range(len(s)):
            if s[i]==c:
                left=i
                right=i
                break
            
        for i in range(len(s)):
            if s[i]==c and left!=i:
                right=i
                break

        for i in range(len(s)):
            if i==right:
                left=right
                for j in range(right+1,len(s)):
                    if s[j]==c:
                        right=j
                        break
        
            nums[i]=min(abs(i-left),abs(right-i))
        return nums
